### ISSUE TITLE
Warn against [ "$(... | wc -l)" -eq 0 ]

### ISSUE THREAD

**Issue by hseg** [koalaman/shellcheck#1793]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
if [ "$(find "$i" -type l | wc -l)" -eq 0 ]; then

```

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

Useless use of `wc`, just write `[ -z "$(cmd)" ]` instead

Admittedly, a bit of a niche use case, but I noticed this misfeature in an old script of mine while cleaning up.

**Comment by dimo414** [comment]
Same for `-ne 0` and `[[ -n ...`, FWIW.
